api/src/main/java/io/strimzi/api/kafka/model/KafkaUserQuotas.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+import io.strimzi.crdgenerator.annotations.Description;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+
+/**
+ * Represent the Quotas configuration for Kafka User
+ */
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@EqualsAndHashCode
+public class KafkaUserQuotas implements UnknownPropertyPreserving, Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private Integer producerByteRate;
+    private Integer consumerByteRate;
+    private Integer requestPercentage;
+
+    private Map<String, Object> additionalProperties;
+
+    @Description("Sets a quota on the maximum bytes per-second that each client group can publish to a broker before the clients in the group are throttled. Defined on a per-broker basis.")
+    public Integer getProducerByteRate() {
+        return producerByteRate;
+    }
+
+    public void setProducerByteRate(Integer producerByteRate) {
+        this.producerByteRate = producerByteRate;
+    }
+
+    @Description("Sets a quota on the maximum bytes per-second that each client group can fetch from a broker before the clients in the group are throttled. Defined on a per-broker basis.")
+    public Integer getConsumerByteRate() {
+        return consumerByteRate;
+    }
+
+    public void setConsumerByteRate(Integer consumerByteRate) {
+        this.consumerByteRate = consumerByteRate;
+    }
+
+    @Description("Sets a quota on the maximum CPU utilization of each client group as a percentage of network and I/O threads.")
+    public Integer getRequestPercentage() {
```

Comment:
Presumably this needs to be in the range 0 to 100? So use `@Minimum` and `@Maximum`. Also, it is really an Integer, or can it be a double in Kafka?

Reply:
I did not see any note about its data type. OTOH it seems it is possible to set it even >100% http://kafka.apache.org/documentation/#quotas

Reply:
From the code it looks like quotas are doubles internally. Please can you verify using `kafka-configs.sh`? Even if >100 is supported, 0 should still be a lower bound.

Reply:
I have tried various combinations but it seems there is no validation. The `kafka-configs` script lets me set int, float but also a negative number. Should we go for the positive floats only?